### DISCUSSION TITLE
ScriptReference/Renderer.SetSharedMaterials

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#540] [Notes]
# ScriptReference/Renderer.SetSharedMaterials

https://docs.unity3d.com/ScriptReference/Renderer.SetSharedMaterials.html

**Comment by PeturDarri** [comment]
There's a bug when you use this to set fewer materials than there currently are on the renderer. The excess materials are not removed, but replaced with the default material.
See the issue tracker [here](https://issuetracker.unity3d.com/issues/removed-material-is-replaced-with-default-material-when-renderer-dot-setmaterials-is-used).
